### PR TITLE
test: added missing tests for the pkg/analysis package

### PR DESCRIPTION
**Pull request by VaibhavMalik4187**
## 📑 Description
This commit adds some of the missing tests in the `pkg/analysis` package and bumps the code coverage to 73.5%

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by VaibhavMalik4187** [comment]
@arbreezy This PR is ready for a code review. Could you please take a look whenever you have a moment? Thanks!

File: pkg/analysis/analysis_test.go [hunk 1]
```diff
@@ -279,3 +289,196 @@ func TestAnalysis_MultipleProblemJsonOutput(t *testing.T) {
...
+			w.Close()
+			got, _ := io.ReadAll(r)
+			os.Stdout = rescueStdout
+
+			require.Equal(t, tt.expectedOutput, string(got))
+		})
+	}
+}
+
+type mockIAI struct {
```
Comment (arbreezy) [inline pkg/analysis/analysis_test.go:459]: Shall we leverage the existing noopAI ?
https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/ai/noopai.go

File: pkg/analysis/output_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+				require.Contains(t, string(output), tt.expectedOutput)
+			} else {
+				require.ErrorContains(t, err, tt.expectedErr)
+				require.Nil(t, output)
+			}
+		})
+	}
+}
+
+func TestJsonOutput(t *testing.T) {
```
Comment (arbreezy) [inline pkg/analysis/output_test.go:67]: Frankly, I am not convinced how valuable is that test, and what we are actually testing here

Comment (JuHyung-Son) [reply pkg/analysis/output_test.go:67]: I agree, I don't know if we need this test either.

File: pkg/analysis/analysis_test.go (hunk 1 above)
Comment (VaibhavMalik4187) [reply pkg/analysis/analysis_test.go:459]: Nice catch, thanks!

File: pkg/analysis/output_test.go (hunk 2 above)
Comment (VaibhavMalik4187) [reply pkg/analysis/output_test.go:67]: True, this test isn't adding some actual value to the test suite. I'll take care of this soon.

Comment (VaibhavMalik4187) [reply pkg/analysis/output_test.go]: Removal complete.

**Review by AlexsJones — APPROVED** [review]

**Review by JuHyung-Son — APPROVED** [review]

**Pull request merged**